### PR TITLE
Move from cbook._warn_external() to _api.warn_external()

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

Followup of moving all API helpers to matplotlib._api (#18657).

Also use more f-strings in the warning messages to make them shorter more readable.

### CODE REVIEW

File: lib/matplotlib/artist.py [hunk 1]
```diff
@@ -901,8 +901,7 @@ def set_rasterized(self, rasterized):
         rasterized : bool
         """
         if rasterized and not hasattr(self.draw, "_supports_rasterization"):
-            cbook._warn_external(
-                "Rasterization of '%s' will be ignored" % self)
+            _api.warn_external("Rasterization of '{self}' will be ignored")
```
Comment (QuLogic) [inline lib/matplotlib/artist.py:904]:
```suggestion
            _api.warn_external(f"Rasterization of '{self}' will be ignored")
```

File: lib/matplotlib/backend_managers.py [hunk 2]
```diff
@@ -437,7 +437,6 @@ def get_tool(self, name, warn=True):
             return name
         if name not in self._tools:
             if warn:
-                cbook._warn_external("ToolManager does not control tool "
-                                     "%s" % name)
+                _api.warn_external("ToolManager does not control tool {name}")
```
Comment (QuLogic) [inline lib/matplotlib/backend_managers.py:440]:
```suggestion
                _api.warn_external(f"ToolManager does not control tool {name}")
```

File: lib/matplotlib/figure.py [hunk 3]
```diff
@@ -3192,12 +3192,12 @@ def execute_constrained_layout(self, renderer=None):
...
         if self._layoutgrid is None:
-            cbook._warn_external("Calling figure.constrained_layout, but "
-                                 "figure not setup to do constrained layout. "
-                                 " You either called GridSpec without the "
-                                 "figure keyword, you are using plt.subplot, "
-                                 "or you need to call figure or subplots "
-                                 "with the constrained_layout=True kwarg.")
+            _api.warn_external("Calling figure.constrained_layout, but "
+                               "figure not setup to do constrained layout. "
+                               " You either called GridSpec without the "
```
Comment (QuLogic) [inline lib/matplotlib/figure.py:3197]:
```suggestion
                               "figure not setup to do constrained layout. "
                               "You either called GridSpec without the "
```

**Review by QuLogic — APPROVED** [review]

**Review by jklymak — APPROVED** [review]
Feel free to self merge after committing @QuLogic suggestion!

**Pull request merged**